internal/aws/client/client.go

```diff
@@ -0,0 +1,65 @@
+package client
+
+import (
+	"context"
+	"github.com/Falco-Talon/falco-talon/utils"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/lambda"
+)
+
+type AWSClient struct {
+	cfg aws.Config
+
+	mu           sync.Mutex // Protects the fields below
+	lambdaClient *lambda.Client
+	//s3Client     *s3.Client
+}
+
+var (
+	awsClient *AWSClient
+	once      sync.Once
+)
+
+func Init() error {
+	var err error
+	once.Do(func() {
+		cfg, err := config.LoadDefaultConfig(context.TODO())
+		if err != nil {
+			return
+		}
+
+		awsClient = &AWSClient{
+			cfg: cfg,
+		}
+	})
+	return err
+}
+
+// GetAWSClient returns the singleton AWSClient instance. Make sure to call Init() before using this function.
+func GetAWSClient() *AWSClient {
+	return awsClient
+}
+
+func (c *AWSClient) LambdaClient() *lambda.Client {
```

Comment:
```suggestion
func (c *AWSClient) GetLambdaClient() *lambda.Client {
```

For the consistency between the names of the functions/methods